### ISSUE TITLE
Global definitions hiding lint errors

### ISSUE THREAD

**Issue by tiagoporto** [microsoft/TypeScript#40333]
My application has to import `.svg` files. When importing an `.svg` **Typescript** report `missing type declarations` error.

[![enter image description here][1]][1]

To handle this error I created a global module declaration

```ts
declare module '*.svg' {
  const content: string
  export default content
}
```

But with this global declaration if the file path is wrong, **TS** doesn't report any errors.

**Example**

Folder structure

```
app
 |
 ├ img
 |  └─ top-view.svg
 |
 └─index.ts

```

index.ts

[![enter image description here][2]][2]

No errors, but the correct path is `../img/top-view.svg`

  [1]: https://i.stack.imgur.com/wPo6E.png
  [2]: https://i.stack.imgur.com/m6IJY.png

**Comment by RyanCavanaugh** [comment]
`declare module` is *intended* to be used when a file with that name isn't present. This is like turning your volume to zero and then expecting to hear music when you press "Play".

**Comment by tiagoporto** [comment]
Hello @RyanCavanaugh 
OK, but if the file path is correct I'm getting the error `Cannot find module` (first image).
If `declare module` is not the correct way to handle not recognized files could you explain how could be do?
Thanks

**Comment by RyanCavanaugh** [comment]
There isn't a mechanism that says "This is how to treat files that are on disk but not TS/JS files"

**Comment by tiagoporto** [comment]
I see, this issue affects many projects. Do you know if in next TS versions there is something in backlog to handle that?
If I could give a suggestion, compiler could still returning `missing type declarations` for not `TS/JS` files. But always check if import path results in a file and return an error if not, even if file is not an `TS/JS`.

**Comment by tiagoporto** [comment]
@RyanCavanaugh Some feedbacks?

**Comment by RyanCavanaugh** [comment]
We don't have any plans right now to add this functionality; this is the first request like this I've seen. You're welcome to log a suggestion issue so that other people can chime in.

**Comment by shaianest** [comment]
dude its not working as intended the intended way is as if you suggest some thing else or always verify import paths 
if you know any way else for module augmentation or any thing els please help others 

example fix might look like "you could inerface it so it can validate path and know what its return looks like
         example interface "*.svg"{
             ReactComponent: JSX.Element
         }

"
